mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/quota/JPAPerUserMaxQuotaDAO.java

```diff
@@ -48,11 +49,11 @@ public JPAPerUserMaxQuotaDAO(EntityManagerFactory entityManagerFactory) {
     }
 
     public void setMaxStorage(QuotaRoot quotaRoot, Optional<QuotaSize> maxStorageQuota) {
-        EntityManager entityManager = entityManagerFactory.createEntityManager();
-        entityManager.getTransaction().begin();
-        MaxUserStorage storedValue = getMaxUserStorageEntity(entityManager, quotaRoot, maxStorageQuota);
-        entityManager.persist(storedValue);
-        entityManager.getTransaction().commit();
+        TransactionRunner.run(entityManagerFactory,
```

Comment:
why don't you inject the TransactionRunner so that you don't have to deal with entityManagerFactory anymore ?